pandas/core/arrays/integer.py

```diff
@@ -345,6 +346,51 @@ def __array__(self, dtype=None):
         """
         return self._coerce_to_ndarray()
 
+    _HANDLED_TYPES = (np.ndarray, numbers.Number)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # For IntegerArray inputs, we apply the ufunc to ._data
+        # and mask the result.
+        out = kwargs.get('out', ())
+
+        for x in inputs + out:
+            if not isinstance(x, self._HANDLED_TYPES + (IntegerArray,)):
+                return NotImplemented
+
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not NotImplemented:
+            return result
+
+        mask = np.zeros(len(self), dtype=bool)
+        inputs2 = []
+        for x in inputs:
+            if isinstance(x, IntegerArray):
+                mask |= x._mask
+                inputs2.append(x._data)
+            else:
+                inputs2.append(x)
+
+        def reconstruct(x):
+            if np.isscalar(x):
+                # reductions.
+                if mask.any():
```

Comment:
The `mask.any()` here means that we follow numpy behaviour and not skip NAs ?

Reply:
I guess so. This affects, e.g. `np.add.reduce`

The current implementing calls the ufunc on `self._data`, including the dummy `1` values for missing values, which makes the value here incorrect when there are missing values.  If we wanted to return a non-NA value here, we would need to handle `.reduce` entirely separate from the rest.

Reply:
In latest numpy there is a ``where`` keyword for this. But yeah, can't really use that yet (unless we decide to raise an error for integerarray for older numpy versions)

Reply:
This 'scalar' branch, is that this relevant if we raise above for 'reduce' methods? (I don't think any of the non-reduce ufuncs can return a scalar